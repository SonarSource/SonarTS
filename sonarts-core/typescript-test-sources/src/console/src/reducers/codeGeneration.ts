import {ReduxAction} from '../types/reducers'
import Constants from '../constants/codeGeneration'
import {Environment, GraphQLClient} from '../types/types'

interface CodeGenerationState {
  environment: Environment
  client: GraphQLClient
}

const initialState: CodeGenerationState = {
  environment: 'Node',
  client: 'lokka',
}

export function reduceCodeGeneration(
  state: CodeGenerationState = initialState,
  action: ReduxAction,
): CodeGenerationState {
  switch (action.type) {
    case Constants.SET_CODE_GENERATION_CLIENT:
      return Object.assign({}, state, {
        client: action.payload,
      })
    case Constants.SET_CODE_GENERATION_ENVIRONMENT:
      return Object.assign({}, state, {
        environment: action.payload,
      })
    default:
      return state
  }
}
