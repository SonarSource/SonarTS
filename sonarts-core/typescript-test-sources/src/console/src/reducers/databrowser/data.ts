import {ReduxAction} from '../../types/reducers'
import Constants from '../../constants/databrowser/data'
import SharedConstants from '../../constants/databrowser/shared'
import * as Immutable from 'immutable'
import {DataBrowserDataState} from '../../types/databrowser/data'

export const initialState: DataBrowserDataState = {
  nodes: Immutable.List<Immutable.Map<string, any>>(),
  backup: {
    nodes: Immutable.List<Immutable.Map<string, any>>(),
    itemCount: 0,
    loaded: Immutable.List<boolean>(),
  },
  orderBy: {
    fieldName: 'id',
    order: 'DESC',
  },
  filter: Immutable.Map<string, any>(),
  itemCount: 0,
  loaded: Immutable.List<boolean>(),
  mutationActive: false,
  newRowShown: false,
  countChanges: Immutable.Map<string, number>(),
}

export function reduceData(state: DataBrowserDataState = initialState, action: ReduxAction): DataBrowserDataState {
  switch (action.type) {
    case Constants.SET_ORDER:
      return Object.assign({}, state, { orderBy: action.payload })
    case Constants.SET_FILTER:
      const {fieldName, value} = action.payload
      return Object.assign({}, state, {
        filter: state.filter.set(fieldName, value),
      })
    case Constants.SET_NEW_ROW_SHOWN:
      return Object.assign({}, state, {
        newRowShown: true,
      })
    case Constants.SET_ITEMCOUNT:
      return Object.assign({}, state, {
        itemCount: action.payload,
      })
    case Constants.SET_DATA:
      const {nodes, loaded} = action.payload
      return Object.assign({}, state, {
        nodes,
        loaded,
      })
    case Constants.SET_NODES:
      return Object.assign({}, state, {
        nodes: action.payload,
      })
    case Constants.SET_LOADED:
      return Object.assign({}, state, {
        loaded: action.payload,
      })
    case Constants.MUTATION_REQUEST:
      return Object.assign({}, state, {
        backup: {
          nodes: state.nodes,
          itemCount: state.itemCount,
          loaded: state.loaded,
        },
        mutationActive: true,
      })
    case Constants.MUTATION_ERROR:
      return Object.assign({}, state, {
        nodes: state.backup.nodes,
        itemCount: state.backup.itemCount,
        loaded: state.backup.loaded,
        mutationActive: false,
      })
    case Constants.MUTATION_SUCCESS:
      return Object.assign({}, state, {
        mutationActive: false,
        backup: {
          nodes: Immutable.List<Immutable.Map<string, any>>(),
          itemCount: 0,
          loaded: Immutable.List<boolean>(),
        },
      })
    case Constants.INCREASE_COUNT_CHANGE:
      return Object.assign({}, state, {
        countChanges: state.countChanges.set(action.payload, (state.countChanges.get(action.payload) || 0) + 1),
      })
    case Constants.DECREASE_COUNT_CHANGE:
      return Object.assign({}, state, {
        countChanges: state.countChanges.set(action.payload, (state.countChanges.get(action.payload) || 0) - 1),
      })
    case Constants.RESET_COUNT_CHANGE:
      return Object.assign({}, state, {
        countChanges: Immutable.Map<string, number>(),
      })
    case Constants.ADD_NODE_REQUEST:
      return Object.assign({}, state, {
        nodes: state.nodes.unshift(action.payload),
        loaded: state.loaded.unshift(true),
        itemCount: state.itemCount + 1,
      })
    case Constants.ADD_NODE_SUCCESS:
      return Object.assign({}, state, {
        nodes: state.nodes.set(0, action.payload),
      })
    case Constants.DELETE_NODES:
      const newState = Object.assign({}, state, {
        nodes: state.nodes.filter((node, index) => action.payload.indexOf(node.get('id')) === -1),
        itemCount: state.itemCount - action.payload.length,
        loaded: state.loaded.slice(0, state.loaded.size - action.payload.length),
      })
      return newState
    case Constants.UPDATE_CELL:
      // use block as we would redefine `value` in the same block again
      const { position } = action.payload
      const cellValue = action.payload.value
      return Object.assign({}, state, {
        nodes: state.nodes.setIn([position.row, position.field], cellValue),
      })
    case SharedConstants.RESET:
      return initialState
  }
  global['state'] = state
  return state
}
