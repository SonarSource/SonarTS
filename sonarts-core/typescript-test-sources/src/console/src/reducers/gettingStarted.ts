import {ReduxAction} from '../types/reducers'
import Constants from '../constants/gettingStarted'
import {GettingStartedState, GettingStartedReducerState} from './../types/gettingStarted'

const initialState: GettingStartedReducerState = {
  poll: false,
  gettingStartedState: new GettingStartedState({
    onboardingStatusId: '',
    selectedExample: null,
    skipped: true,
    step: 'STEP6_CLOSED',
  }),
}

export function reduceGettingStartedState(state: GettingStartedReducerState = initialState, action: ReduxAction): GettingStartedReducerState { // tslint:disable-line
  switch (action.type) {
    case Constants.UPDATE:
      const gettingStartedState: GettingStartedState = action.payload.gettingStartedState
      if (!gettingStartedState.selectedExample) {
        gettingStartedState.selectedExample = state.gettingStartedState.selectedExample
      }
      // TODO: use reselect for derived data
      return Object.assign({}, state, {
        gettingStartedState,
        poll: gettingStartedState.isCurrentStep('STEP5_WAITING'),
      })
    default:
      return state
  }
}
