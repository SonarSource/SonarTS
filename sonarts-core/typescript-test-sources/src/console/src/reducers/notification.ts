import { ReduxAction } from '../types/reducers'
import Constants from '../constants/notification'
import { NotificationState } from '../types/notification'

const initialState: NotificationState = {
  message: null,
  level: null,
}

export function reduceNotification(state: NotificationState = initialState, action: ReduxAction): NotificationState {
  switch (action.type) {
    case Constants.SHOW_NOTIFICATION:
      return {
        message: action.payload.message,
        level: action.payload.level,
      }
    case Constants.CLEAR_NOTIFICATION:
      return {
        message: null,
        level: null,
      }
    default:
      return state
  }
}
