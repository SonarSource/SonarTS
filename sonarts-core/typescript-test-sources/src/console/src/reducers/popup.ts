import {ReduxAction} from '../types/reducers'
import Constants from '../constants/popup'
import {Popup, PopupState} from '../types/popup'
import * as Immutable from 'immutable'

const initialState: PopupState = {
  popups: Immutable.List<Popup>(),
}

export function reducePopup(state: PopupState = initialState, action: ReduxAction): PopupState {
  switch (action.type) {
    case Constants.SHOW_POPUP:
      return {
        popups: state.popups.push(action.payload),
      }
    case Constants.CLOSE_POPUP:
      if (!action.payload) {
        return {
          popups: state.popups.pop(),
        }
      }
      return {
        popups: state.popups.filter(value => value.id !== action.payload).toList(),
      }
    default:
      return state
  }
}
