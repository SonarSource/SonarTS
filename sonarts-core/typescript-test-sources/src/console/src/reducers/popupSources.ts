import {ReduxAction} from '../types/reducers'
import Constants from '../constants/popupSources'
import {PopupSourcesState} from '../types/popupSources'

const initialState: PopupSourcesState = {
  relationsPopup: 'schema',
  fieldPopup: 'schema',
}

export default function reduce(state: PopupSourcesState = initialState, action: ReduxAction): PopupSourcesState {
  switch (action.type) {
    case Constants.SET_FIELD_POPUP_SOURCE:
      return Object.assign({}, state, {
        fieldPopup: action.payload,
      })
    case Constants.SET_RELATIONS_POPUP_SOURCE:
      return Object.assign({}, state, {
        relationsPopup: action.payload,
      })
    default:
      return state
  }
}
