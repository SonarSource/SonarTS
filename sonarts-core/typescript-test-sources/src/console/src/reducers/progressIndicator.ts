import {ReduxAction} from '../types/reducers'
import Constants from '../constants/progressIndicator'
import {ProgressIndicatorState} from '../types/progressIndicator'

const initialState: ProgressIndicatorState = {
  progress: 0,
}

export function reduceProgress (state: ProgressIndicatorState = initialState, action: ReduxAction): ProgressIndicatorState { // tslint:disable-line
  switch (action.type) {
    case Constants.START_PROGRESS:
      return Object.assign({}, initialState)
    case Constants.INCREMENT_PROGRESS:
      return Object.assign({}, {
        progress: state.progress + 1,
      })
    default:
      return state
  }
}
