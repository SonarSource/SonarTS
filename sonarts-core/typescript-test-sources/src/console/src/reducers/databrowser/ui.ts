import {ReduxAction} from '../../types/reducers'
import Constants from '../../constants/databrowser/ui'
import SharedConstants from '../../constants/databrowser/shared'
import * as Immutable from 'immutable'
import {DataBrowserUIState} from '../../types/databrowser/ui'
import {ActionRowState} from '../../types/databrowser/actionrow'

const initialState: DataBrowserUIState = {
  searchVisible: false,
  newRowActive: false,
  selectedNodeIds: Immutable.List<string>(),
  scrollTop: 0,
  loading: true,
  writing: false,
  actionRow: ActionRowState.NewNode,
  selectedCell: {
    row: -1,
    field: null,
  },
  editing: false,
  searchQuery: null,
}

export function reduceUI(state: DataBrowserUIState = initialState, action: ReduxAction): DataBrowserUIState {
  switch (action.type) {
    case Constants.HIDE_NEW_ROW:
      return Object.assign({}, state, {
        newRowActive: false,
      })
    case Constants.FORCE_SHOW_NEW_ROW:
      return Object.assign({}, state, {
        newRowActive: true,
      })
    case Constants.TOGGLE_NEW_ROW:
      return Object.assign({}, state, {
        newRowActive: !state.newRowActive,
      })
    case Constants.TOGGLE_SEARCH:
      return Object.assign({}, state, {
        searchVisible: !state.searchVisible,
      })
    case Constants.SET_NODE_SELECTION:
      return Object.assign({}, state, {
        selectedNodeIds: action.payload,
        actionRow: action.payload.size > 0 ? ActionRowState.DeleteNode : ActionRowState.NewNode,
      })
    case Constants.CLEAR_NODE_SELECTION:
      return Object.assign({}, state, {
        selectedNodeIds: Immutable.List<string>(),
        actionRow: ActionRowState.NewNode,
      })
    case Constants.SELECT_CELL:
      return Object.assign({}, state, {
        selectedCell: action.payload,
      })
    case Constants.UNSELECT_CELL:
      return Object.assign({}, state, {
        selectedCell: {
          row: -1,
          field: null,
        },
      })
    case Constants.EDIT_CELL:
      return Object.assign({}, state, {
        editing: true,
        selectedCell: action.payload,
      })
    case Constants.STOP_EDIT_CELL:
      return Object.assign({}, state, {
        editing: false,
      })
    case Constants.TOGGLE_NODE_SELECTION:
      const id = action.payload
      if (state.selectedNodeIds.includes(id)) {
        const nodes = state.selectedNodeIds.filter(x => x !== id)
        return Object.assign({}, state, {
          selectedNodeIds: nodes,
          actionRow: nodes.size > 0 ? ActionRowState.DeleteNode : ActionRowState.NewNode,
        })
      }
      return Object.assign({}, state, {
        selectedNodeIds: state.selectedNodeIds.push(id), // using Immutable.js push here
        actionRow: ActionRowState.DeleteNode,
      })
    case Constants.SET_SCROLL_TOP:
      return Object.assign({}, state, {
        scrollTop: action.payload,
      })
    case Constants.SET_LOADING:
      return Object.assign({}, state, {
        loading: action.payload,
      })
    case Constants.SET_WRITING:
      return Object.assign({}, state, {
        writing: action.payload,
      })
    case Constants.ACTIVATE_NEW_NODE_ROW:
      return Object.assign({}, state, {
        actionRow: ActionRowState.NewNode,
      })
    case Constants.ACTIVATE_SEARCH_ROW:
      return Object.assign({}, state, {
        actionRow: ActionRowState.Search,
      })
    case Constants.ACTIVATE_DELETE_NODE_ROW:
      return Object.assign({}, state, {
        actionRow: ActionRowState.DeleteNode,
      })
    case SharedConstants.RESET:
      return initialState
  }
  return state
}
