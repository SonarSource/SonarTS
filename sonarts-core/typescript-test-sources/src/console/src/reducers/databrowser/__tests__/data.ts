import { reduceData, initialState } from '../data'
import {
  setOrder, setItemCount, setData, mutationRequest,
  addNodeRequest, mutationError, deleteNodes, mutationSuccess, updateCell, addNodeSuccess,
} from '../../../actions/databrowser/data'
import * as Immutable from 'immutable'
import {ReduxAction} from '../../../types/reducers'
import {DataBrowserDataState} from '../../../types/databrowser/data'
import {Reducer} from 'redux'

function mockNodes() {
  return setData(
    Immutable.List<Immutable.Map<string, any>>(
      [
        Immutable.Map({
          id: 'asd',
          name: 'a',
        }),
        Immutable.Map({
          id: 'asd2',
          name: 'a2',
        }),
        Immutable.Map({
          id: 'asd3',
          name: 'a3',
        }),
      ],
    ),
    Immutable.List<boolean>([true, true, true]),
  )
}

describe('databrowser reducer', () => {

  it('should render the initialState', () => {
    const state = reduceData(initialState, {type: null})
    expect(state).toMatchSnapshot()
  })

  it('should set the order', () => {
    const state = reduceData(initialState, setOrder({
      fieldName: 'name',
      order: 'ASC',
    }))

    expect(state).toMatchSnapshot()
  })

  it('should set the item count', () => {
    const state = reduceData(initialState, setItemCount(5))

    expect(state).toMatchSnapshot()
  })

  it('should set data', () => {
    const state = reduceData(initialState, mockNodes())

    expect(state).toMatchSnapshot()
  })

  it('should backup and restore nodes on create mutation request', () => {
    reduceAndSnapshot(initialState, reduceData, [
      mockNodes(),
      mutationRequest(),
      addNodeRequest(Immutable.Map({
        id: 'asd4',
        name: 'a4',
      })),
      mutationError(),
    ])
  })

  it('should be able to modify the created node after optimistic create', () => {
    reduceAndSnapshot(
      initialState,
      reduceData,
      [
        mockNodes(),
        mutationRequest(),
        addNodeRequest(Immutable.Map({
          name: 'a4',
        })),
        addNodeSuccess(Immutable.Map({
          id: 'asd4',
          name: 'a4',
        })),
      ],
      [0, 1],
    )
  })

  it('should backup and delete the backup when mutation success after delete mutation', () => {
    reduceAndSnapshot(
      initialState,
      reduceData,
      [
        mockNodes(),
        mutationRequest(),
        deleteNodes(['asd', 'asd2']),
        mutationSuccess(),
      ],
      [0, 1])
  })

  it('should backup and restore when mutation fail after delete mutation', () => {
    reduceAndSnapshot(
      initialState,
      reduceData,
      [
        mockNodes(),
        mutationRequest(),
        deleteNodes(['asd', 'asd2']),
        mutationError(),
      ],
      [0, 1],
    )
  })

  it('should execute the update cell action', () => {
    reduceAndSnapshot(initialState, reduceData, [
      mockNodes(),
      updateCell({
        position: {
          row: 0,
          field: 'name',
        },
        value: 'aaa',
      }),
    ])
  })

})

function reduceAndSnapshot(
  initialState: DataBrowserDataState,
  reduce: Reducer<DataBrowserDataState>,
  actions: ReduxAction[],
  excludeActions: number[] = [],
) {
  let lastState = initialState
  actions.forEach((action, index) => {
    lastState = reduce(lastState, action)
    if (excludeActions.includes(index)) {
      return
    }
    expect(lastState).toMatchSnapshot()
  })
}
