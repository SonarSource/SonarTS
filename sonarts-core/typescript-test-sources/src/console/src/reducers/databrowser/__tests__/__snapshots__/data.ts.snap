exports[`databrowser reducer should backup and delete the backup when mutation success after delete mutation 1`] = `
Object {
  "backup": Object {
    "itemCount": 0,
    "loaded": Array [
      true,
      true,
      true,
    ],
    "nodes": Array [
      Object {
        "id": "asd",
        "name": "a",
      },
      Object {
        "id": "asd2",
        "name": "a2",
      },
      Object {
        "id": "asd3",
        "name": "a3",
      },
    ],
  },
  "countChanges": Object {},
  "filter": Object {},
  "itemCount": -2,
  "loaded": Array [
    true,
  ],
  "mutationActive": true,
  "newRowShown": false,
  "nodes": Array [
    Object {
      "id": "asd3",
      "name": "a3",
    },
  ],
  "orderBy": Object {
    "fieldName": "id",
    "order": "DESC",
  },
}
`;

exports[`databrowser reducer should backup and delete the backup when mutation success after delete mutation 2`] = `
Object {
  "backup": Object {
    "itemCount": 0,
    "loaded": Array [],
    "nodes": Array [],
  },
  "countChanges": Object {},
  "filter": Object {},
  "itemCount": -2,
  "loaded": Array [
    true,
  ],
  "mutationActive": false,
  "newRowShown": false,
  "nodes": Array [
    Object {
      "id": "asd3",
      "name": "a3",
    },
  ],
  "orderBy": Object {
    "fieldName": "id",
    "order": "DESC",
  },
}
`;

exports[`databrowser reducer should backup and restore nodes on create mutation request 1`] = `
Object {
  "backup": Object {
    "itemCount": 0,
    "loaded": Array [],
    "nodes": Array [],
  },
  "countChanges": Object {},
  "filter": Object {},
  "itemCount": 0,
  "loaded": Array [
    true,
    true,
    true,
  ],
  "mutationActive": false,
  "newRowShown": false,
  "nodes": Array [
    Object {
      "id": "asd",
      "name": "a",
    },
    Object {
      "id": "asd2",
      "name": "a2",
    },
    Object {
      "id": "asd3",
      "name": "a3",
    },
  ],
  "orderBy": Object {
    "fieldName": "id",
    "order": "DESC",
  },
}
`;

exports[`databrowser reducer should backup and restore nodes on create mutation request 2`] = `
Object {
  "backup": Object {
    "itemCount": 0,
    "loaded": Array [
      true,
      true,
      true,
    ],
    "nodes": Array [
      Object {
        "id": "asd",
        "name": "a",
      },
      Object {
        "id": "asd2",
        "name": "a2",
      },
      Object {
        "id": "asd3",
        "name": "a3",
      },
    ],
  },
  "countChanges": Object {},
  "filter": Object {},
  "itemCount": 0,
  "loaded": Array [
    true,
    true,
    true,
  ],
  "mutationActive": true,
  "newRowShown": false,
  "nodes": Array [
    Object {
      "id": "asd",
      "name": "a",
    },
    Object {
      "id": "asd2",
      "name": "a2",
    },
    Object {
      "id": "asd3",
      "name": "a3",
    },
  ],
  "orderBy": Object {
    "fieldName": "id",
    "order": "DESC",
  },
}
`;

exports[`databrowser reducer should backup and restore nodes on create mutation request 3`] = `
Object {
  "backup": Object {
    "itemCount": 0,
    "loaded": Array [
      true,
      true,
      true,
    ],
    "nodes": Array [
      Object {
        "id": "asd",
        "name": "a",
      },
      Object {
        "id": "asd2",
        "name": "a2",
      },
      Object {
        "id": "asd3",
        "name": "a3",
      },
    ],
  },
  "countChanges": Object {},
  "filter": Object {},
  "itemCount": 1,
  "loaded": Array [
    true,
    true,
    true,
    true,
  ],
  "mutationActive": true,
  "newRowShown": false,
  "nodes": Array [
    Object {
      "id": "asd4",
      "name": "a4",
    },
    Object {
      "id": "asd",
      "name": "a",
    },
    Object {
      "id": "asd2",
      "name": "a2",
    },
    Object {
      "id": "asd3",
      "name": "a3",
    },
  ],
  "orderBy": Object {
    "fieldName": "id",
    "order": "DESC",
  },
}
`;

exports[`databrowser reducer should backup and restore nodes on create mutation request 4`] = `
Object {
  "backup": Object {
    "itemCount": 0,
    "loaded": Array [
      true,
      true,
      true,
    ],
    "nodes": Array [
      Object {
        "id": "asd",
        "name": "a",
      },
      Object {
        "id": "asd2",
        "name": "a2",
      },
      Object {
        "id": "asd3",
        "name": "a3",
      },
    ],
  },
  "countChanges": Object {},
  "filter": Object {},
  "itemCount": 0,
  "loaded": Array [
    true,
    true,
    true,
  ],
  "mutationActive": false,
  "newRowShown": false,
  "nodes": Array [
    Object {
      "id": "asd",
      "name": "a",
    },
    Object {
      "id": "asd2",
      "name": "a2",
    },
    Object {
      "id": "asd3",
      "name": "a3",
    },
  ],
  "orderBy": Object {
    "fieldName": "id",
    "order": "DESC",
  },
}
`;

exports[`databrowser reducer should backup and restore when mutation fail after delete mutation 1`] = `
Object {
  "backup": Object {
    "itemCount": 0,
    "loaded": Array [
      true,
      true,
      true,
    ],
    "nodes": Array [
      Object {
        "id": "asd",
        "name": "a",
      },
      Object {
        "id": "asd2",
        "name": "a2",
      },
      Object {
        "id": "asd3",
        "name": "a3",
      },
    ],
  },
  "countChanges": Object {},
  "filter": Object {},
  "itemCount": -2,
  "loaded": Array [
    true,
  ],
  "mutationActive": true,
  "newRowShown": false,
  "nodes": Array [
    Object {
      "id": "asd3",
      "name": "a3",
    },
  ],
  "orderBy": Object {
    "fieldName": "id",
    "order": "DESC",
  },
}
`;

exports[`databrowser reducer should backup and restore when mutation fail after delete mutation 2`] = `
Object {
  "backup": Object {
    "itemCount": 0,
    "loaded": Array [
      true,
      true,
      true,
    ],
    "nodes": Array [
      Object {
        "id": "asd",
        "name": "a",
      },
      Object {
        "id": "asd2",
        "name": "a2",
      },
      Object {
        "id": "asd3",
        "name": "a3",
      },
    ],
  },
  "countChanges": Object {},
  "filter": Object {},
  "itemCount": 0,
  "loaded": Array [
    true,
    true,
    true,
  ],
  "mutationActive": false,
  "newRowShown": false,
  "nodes": Array [
    Object {
      "id": "asd",
      "name": "a",
    },
    Object {
      "id": "asd2",
      "name": "a2",
    },
    Object {
      "id": "asd3",
      "name": "a3",
    },
  ],
  "orderBy": Object {
    "fieldName": "id",
    "order": "DESC",
  },
}
`;

exports[`databrowser reducer should be able to modify the created node after optimistic create 1`] = `
Object {
  "backup": Object {
    "itemCount": 0,
    "loaded": Array [
      true,
      true,
      true,
    ],
    "nodes": Array [
      Object {
        "id": "asd",
        "name": "a",
      },
      Object {
        "id": "asd2",
        "name": "a2",
      },
      Object {
        "id": "asd3",
        "name": "a3",
      },
    ],
  },
  "countChanges": Object {},
  "filter": Object {},
  "itemCount": 1,
  "loaded": Array [
    true,
    true,
    true,
    true,
  ],
  "mutationActive": true,
  "newRowShown": false,
  "nodes": Array [
    Object {
      "name": "a4",
    },
    Object {
      "id": "asd",
      "name": "a",
    },
    Object {
      "id": "asd2",
      "name": "a2",
    },
    Object {
      "id": "asd3",
      "name": "a3",
    },
  ],
  "orderBy": Object {
    "fieldName": "id",
    "order": "DESC",
  },
}
`;

exports[`databrowser reducer should be able to modify the created node after optimistic create 2`] = `
Object {
  "backup": Object {
    "itemCount": 0,
    "loaded": Array [
      true,
      true,
      true,
    ],
    "nodes": Array [
      Object {
        "id": "asd",
        "name": "a",
      },
      Object {
        "id": "asd2",
        "name": "a2",
      },
      Object {
        "id": "asd3",
        "name": "a3",
      },
    ],
  },
  "countChanges": Object {},
  "filter": Object {},
  "itemCount": 1,
  "loaded": Array [
    true,
    true,
    true,
    true,
  ],
  "mutationActive": true,
  "newRowShown": false,
  "nodes": Array [
    Object {
      "id": "asd4",
      "name": "a4",
    },
    Object {
      "id": "asd",
      "name": "a",
    },
    Object {
      "id": "asd2",
      "name": "a2",
    },
    Object {
      "id": "asd3",
      "name": "a3",
    },
  ],
  "orderBy": Object {
    "fieldName": "id",
    "order": "DESC",
  },
}
`;

exports[`databrowser reducer should execute the update cell action 1`] = `
Object {
  "backup": Object {
    "itemCount": 0,
    "loaded": Array [],
    "nodes": Array [],
  },
  "countChanges": Object {},
  "filter": Object {},
  "itemCount": 0,
  "loaded": Array [
    true,
    true,
    true,
  ],
  "mutationActive": false,
  "newRowShown": false,
  "nodes": Array [
    Object {
      "id": "asd",
      "name": "a",
    },
    Object {
      "id": "asd2",
      "name": "a2",
    },
    Object {
      "id": "asd3",
      "name": "a3",
    },
  ],
  "orderBy": Object {
    "fieldName": "id",
    "order": "DESC",
  },
}
`;

exports[`databrowser reducer should execute the update cell action 2`] = `
Object {
  "backup": Object {
    "itemCount": 0,
    "loaded": Array [],
    "nodes": Array [],
  },
  "countChanges": Object {},
  "filter": Object {},
  "itemCount": 0,
  "loaded": Array [
    true,
    true,
    true,
  ],
  "mutationActive": false,
  "newRowShown": false,
  "nodes": Array [
    Object {
      "id": "asd",
      "name": "aaa",
    },
    Object {
      "id": "asd2",
      "name": "a2",
    },
    Object {
      "id": "asd3",
      "name": "a3",
    },
  ],
  "orderBy": Object {
    "fieldName": "id",
    "order": "DESC",
  },
}
`;

exports[`databrowser reducer should render the initialState 1`] = `
Object {
  "backup": Object {
    "itemCount": 0,
    "loaded": Array [],
    "nodes": Array [],
  },
  "countChanges": Object {},
  "filter": Object {},
  "itemCount": 0,
  "loaded": Array [],
  "mutationActive": false,
  "newRowShown": false,
  "nodes": Array [],
  "orderBy": Object {
    "fieldName": "id",
    "order": "DESC",
  },
}
`;

exports[`databrowser reducer should set data 1`] = `
Object {
  "backup": Object {
    "itemCount": 0,
    "loaded": Array [],
    "nodes": Array [],
  },
  "countChanges": Object {},
  "filter": Object {},
  "itemCount": 0,
  "loaded": Array [
    true,
    true,
    true,
  ],
  "mutationActive": false,
  "newRowShown": false,
  "nodes": Array [
    Object {
      "id": "asd",
      "name": "a",
    },
    Object {
      "id": "asd2",
      "name": "a2",
    },
    Object {
      "id": "asd3",
      "name": "a3",
    },
  ],
  "orderBy": Object {
    "fieldName": "id",
    "order": "DESC",
  },
}
`;

exports[`databrowser reducer should set the item count 1`] = `
Object {
  "backup": Object {
    "itemCount": 0,
    "loaded": Array [],
    "nodes": Array [],
  },
  "countChanges": Object {},
  "filter": Object {},
  "itemCount": 5,
  "loaded": Array [],
  "mutationActive": false,
  "newRowShown": false,
  "nodes": Array [],
  "orderBy": Object {
    "fieldName": "id",
    "order": "DESC",
  },
}
`;

exports[`databrowser reducer should set the order 1`] = `
Object {
  "backup": Object {
    "itemCount": 0,
    "loaded": Array [],
    "nodes": Array [],
  },
  "countChanges": Object {},
  "filter": Object {},
  "itemCount": 0,
  "loaded": Array [],
  "mutationActive": false,
  "newRowShown": false,
  "nodes": Array [],
  "orderBy": Object {
    "fieldName": "name",
    "order": "ASC",
  },
}
`;
